clang/lib/Sema/SemaExpr.cpp

```diff
@@ -229,6 +229,20 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                !checkAllowedSYCLInitializer(VD, /*CheckValueDependent =*/true))
         SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
             << Sema::KernelConstStaticVariable;
+      else if (IsConst && VD->hasGlobalStorage()) {
+        QualType Ty = VD->getType();
+        if (const auto *CAType = dyn_cast<ConstantArrayType>(Ty)) {
+          assert(CAType);
+          // Max length of SPIRV instruction is 65535 words.
+          // First 3 words of OpConstantComposite encode: 1) word count &
```

Comment:
What is OpConstantComposite?

Reply:
OpConstantComposite is a SPIR-V instruction to declare a new array/vector/matrix/structure constant.
For, `vec4 scale = vec4(1.0, 1.0, 2.0, 1.0);` - corresponding SPIR-V will be:
```
%7 = OpTypeVector
%10 = OpConstant  1.0
%11 = OpConstant  2.0
%12 = OpConstantComposite %7 %10 %10 %11 %10 ; // vec4(1.0, 1.0, 2.0, 1.0)
```